CHANGELOG.md

```diff
@@ -5,6 +5,8 @@
 * [FEATURE] Query Frontend: Add dynamic interval size for query splitting. This is enabled by configuring experimental flags `querier.max-shards-per-query` and/or `querier.max-fetched-data-duration-per-query`. The split interval size is dynamically increased to maintain a number of shards and total duration fetched below the configured values. #6458
 * [FEATURE] Querier/Ruler: Add `query_partial_data` and `rules_partial_data` limits to allow queries/rules to be evaluated with data from a single zone, if other zones are not available. #6526
 * [FEATURE] Update prometheus alertmanager version to v0.28.0 and add new integration msteamsv2, jira, and rocketchat. #6590
+* [FEATURE] Ingester: Add a `-ingester.enable-ooo-native-histograms` flag to enable out-of-order native histogram ingestion per tenant. It only takes effect when `-blocks-storage.tsdb.enable-native-histograms=true` and `-ingester.out-of-order-time-window` > 0. It is applied after the restart if it is changed at runtime through the runtime config. #6626
+* [FEATURE] Ingester/StoreGateway: Add `resource-thresholds` in ingesters and store gateways to throttle query requests when the pods are under resource pressure. #6674
```

Comment:
Maybe it would be interesting to know/document the use-case exactly? Is it so that the user would get a nice error message instead of waiting for the timeout to hit when there's a resource pressure?

Reply:
What we've noticed is that customers often make a mistake on their query string, step or period, making unintentional heavy queries and killing our componets.

Here's some use-cases I thought of:
1. Heavy queries on ingesters spike up CPU, impacting RemoteWrite
2. Heavy queries on query components OOMKilling the components

If I were to add a documentation with examples, is there a standard format or place to add?

Reply:
Added a new doc `protecting-cortex-from-heavy-queries.md` under guides